security/cmd/node_agent/management/server.go

```diff
@@ -158,12 +160,14 @@ func (s *Server) WorkloadAdded(ctx context.Context, request *pb.WorkloadInfo) (*
 	if err != nil {
 		return nil, fmt.Errorf("csr request failed for service %v %v", sa, err)
 	}
-
-	if err := s.ss.SetServiceIdentityPrivateKey(priv); err != nil {
-		return nil, fmt.Errorf("failed in set private key for service account %v, uid %v err %v", sa, uid, err)
+	// TODO: get root somewhere?
+	kb, err := pkiutil.NewVerifiedKeyCertBundleFromPem(resp.SignedCert, priv, resp.CertChain, nil)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build key cert bundle")
 	}
-	if err := s.ss.SetServiceIdentityCert(resp.SignedCert); err != nil {
-		return nil, fmt.Errorf("failed in set cert for service account %v, uid %v err %v", sa, uid, err)
+
+	if err := s.ss.Save(kb); err != nil {
```

Comment:
I feel ss is not very readable. Expand it?

Reply:
Changed to secretServer